scale/codec.go

```diff
@@ -487,51 +482,60 @@ func (pd Decoder) DecodeIntoReflectValue(target reflect.Value) error {
 }
 
 // DecodeUintCompact decodes a compact-encoded integer. See EncodeUintCompact method.
-func (pd Decoder) DecodeUintCompact() (uint64, error) {
+func (pd Decoder) DecodeUintCompact() (*big.Int, error) {
```

Comment:
Why do we return a pointer to a `big.Int` here, and not a `big.Int`?

Reply:
I am attracted to the idea of returning nil if there is an error in the call and the value ref vs copy. If you have strong arguments against it let me know.